lib/runtimeSettings.js

```diff
@@ -156,6 +157,12 @@ function getSettingsFile (settings) {
         }
     }
 
+    // all current drivers add settings.rootDir and settings.userDir
+    if (settings.rootDir) {
+        const uibRoot = path.join(settings.rootDir, settings.userDir, 'storage', 'uibuilder')
+        projectSettings.uiBuilder = `uibuilder: { uibRoot: '${uibRoot}' },`
```

Comment:
This wont handle windows paths and pretty certain any file path with a quote will break the generated settings file

![image](https://github.com/FlowFuse/nr-launcher/assets/44235289/02a55384-c6d8-4684-94d5-6ad6cabe88fd)

I think it might be necessary to convert to posix paths and just JSON stringify to remove any doubt or ambiguity

Reply:
to clarify, the `\\` will get written as a single `\` and when imported I think the `\` will escape the `S` of `Steve`

Reply:
@Steve-Mcl paths will never have quotes in the middle (unless some unhinged mad man puts them there, in which case they deserve all they get), paths with spaces get quoted as a whole, e.g. `"C:\Foo Bar"`

Can we not make use of `path.normalize` here and https://nodejs.org/docs/latest-v20.x/api/path.html#windows-vs-posix

Reply:
Update applied.